File: springboot-mall/src/main/java/com/tonyydl/springbootmall/service/UserService.java
package com.tonyydl.springbootmall.service;

import com.tonyydl.springbootmall.data.dto.UserLoginRequestDTO;
import com.tonyydl.springbootmall.data.dto.UserRegisterRequestDTO;
import com.tonyydl.springbootmall.data.po.UserPO;

public interface UserService {
    UserPO getUserById(Integer userId);

    Integer register(UserRegisterRequestDTO userRegisterRequestDTO);

    UserPO login(UserLoginRequestDTO userLoginRequestDTO);
}

File: android-app/app/src/main/java/com/tonyydl/shoppingmallapp/service/UserService.kt
package com.tonyydl.shoppingmallapp.service

import com.tonyydl.shoppingmallapp.data.dto.UserDTO
import com.tonyydl.shoppingmallapp.data.dto.UserLoginRequestDTO
import retrofit2.http.Body
import retrofit2.http.POST

interface UserService {

    @POST("/users/login")
    suspend fun login(@Body userLoginRequestDTO: UserLoginRequestDTO): UserDTO
}